lib/functions.inc.php

```diff
@@ -239,6 +260,29 @@ function check_password_strength( $password, $oldpassword, $pwd_policy_config, $
     return $result;
 }
 
+function check_sshkey( $ssh_pubkey, $sshkey_policy_config ) {
+    extract( $sshkey_policy_config );
+
+    # Split up the SSH key
+    $parts = explode(' ', $ssh_pubkey);
+
+    # Check this gave us at least two parts, and SSH key type is valid
+    if (count($parts) < 2 || $parts[0] != 'ssh-rsa') { return "keyinvalid"; }
```

Comment:
The function considers all non rsa keys as invalid. I do not think it is acceptable to disallow other types of keys : dsa, ecdsa and ed25519.